### PR TITLE
Error/trace correlation model

### PR DESCRIPTION
**Pull request by lemnik**
## Goal
Allow the `Event` model to include a performance trace/span ID for error correlation.

## Testing
Added new tests to ensure the new fields are correctly serialized / deserialized from JSON

### CODE REVIEW

File: bugsnag-android-core/src/main/java/com/bugsnag/android/BugsnagEventMapper.kt [hunk 1]
```diff
@@ -93,6 +94,16 @@ internal class BugsnagEventMapper(
         // populate internalMetrics
         event.internalMetrics = InternalMetricsImpl(map["usage"] as MutableMap<String, Any>?)
 
+        // populate correlation
+        (map["correlation"] as? Map<String, String>)?.let {
+            val traceId = parseTraceId(it["traceid"])
+            val spanId = it["spanid"]?.parseUnsignedLong()
+
+            if (traceId != null && spanId != null) {
```
[the remark this answers is not in the document]
Comment (tomlongridge) [reply bugsnag-android-core/src/main/java/com/bugsnag/android/BugsnagEventMapper.kt:102]: Discussed IRL and agreed that this should be all or nothing and the spanId can always be set to `0`.

**Review by tomlongridge — APPROVED** [review]

**Review by YYChen01988 — APPROVED** [review]
LGTM

File: bugsnag-android-core/src/main/java/com/bugsnag/android/TraceCorrelation.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package com.bugsnag.android
+
+import java.util.UUID
+
+internal data class TraceCorrelation(val traceId: UUID, val spanId: Long) : JsonStream.Streamable {
+    override fun toStream(writer: JsonStream) {
+        writer.beginObject()
+            .name("traceid").value(traceId.toHexString())
+            .name("spanid").value(spanId.toHexString())
```
Comment (tomlongridge) [inline bugsnag-android-core/src/main/java/com/bugsnag/android/TraceCorrelation.kt:9]: As per discussion, please now camelcase these.

Comment (lemnik) [reply bugsnag-android-core/src/main/java/com/bugsnag/android/TraceCorrelation.kt]: done 👍

**Pull request merged**